### DISCUSSION TITLE
Misconfiguration on Magick Mogrify

### DISCUSSION THREAD

**Discussion by Vanir-Labs** [ImageMagick/ImageMagick#6617] [Help] [answered]
### ImageMagick version

7.1.1.15 Q16-HDRI 64x

### Operating system, version and so on

Windows 11 64x

### Description

Hello. I'm trying to convert these images after I ran a colour correction on them. However, when I do, it keeps changing my bit depth and transparency layer every time it's processed. Am I misunderstanding something?
My images must be 24 bit depth, RGB, no Transparencies, 256colour max, and 24-BMP or else it won't work.

My commands look like:
`
magick mogrify "C:\Users\texture\holiday.bmp" -depth 24 -alpha off "C:\Users\texture\holiday.bmp"
`
And another example
`
magick "C:\Users\texture\holiday.bmp" -type truecolor +dither -colors 256 -depth 24 -alpha off "C:\Users\texture\holiday.bmp"
`

**Comment by snibgo** [comment]
"`-depth 24`" is wrong. I suppose you want `-depth 8`, the number of bits per channel per pixel. It isn't the total number of bits per pixel. If there are 3 channels, the total bits per pixel would be 24.

The "`magick mogrify`" command is wrong. See https://imagemagick.org/script/mogrify.php for the syntax. It usually takes a single filename, at the end, for the input and output filename, which can include wildcards.

If you want to reduce the number of colours to 256, and you want to ignore alpha, I suggest ignoring alpha *first*.

IM will try to save in the smallest BMP format. With 256 or fewer colours, the output will be a palette file. Use "`-type truecolor`" to make it 3-channel, ie 24 bits total per pixel.
```
magick in.png -alpha off -colors 256 -type truecolor out.bmp
```

**Reply by Vanir-Labs** [reply]
Thanks, I did have depth 8 before but my images keep saying depth 24, so I'm not sure if it's working or waht. I will try this command you posted now.

**Reply by Vanir-Labs** [reply]
So it changed everything, assumingly to what it said. However, the Bit Depts is 24 again now.
My original file has 8BD 
![image](https://github.com/ImageMagick/ImageMagick/assets/82656075/21017ee0-35ac-402e-9003-263cc6e84bdf)

The new one has 24. How do I keep this, or change this, to 8 BD with 24BMP after this?
![image](https://github.com/ImageMagick/ImageMagick/assets/82656075/d8c5a413-5c69-4939-b1e9-4851cdd591f9)

**Reply by fmw42** [reply]
Windows may be reporting the total bit depth per pixel, but IM uses bit depth per channel per pixel.

**Reply by Vanir-Labs** [reply]
Whatever Windows is telling me, seems to be working though. The 8bd shows in the game but the 24bd shows as solid black in it.

**Comment by snibgo** [comment, marked as answer]
The word "depth" means either "bits per channel per pixel" or "total bits per pixel". Different software uses different definitions.

In IM, it means "bits per channel per pixel".

BMP has no format of 24 bits per channel per pixel. Therefore, any software that says "depth 24" means "total 24 bits per pixel". If that is what you want, then use "`-type truecolor`".

If you want a BMP file with 8 bits total per pixel, that would be "`-type palette`".

**Comment by Vanir-Labs** [comment]
All the answers were correct. The issue I had was actually a BMP version problem. I needed BMP3 and it was defaulted to 4. Thanks for the help!
